From 8ebbd6b54586912a7c5ce75518e9b91dd18393e8 Mon Sep 17 00:00:00 2001
From: Charles Stevens <charles.stevens@compasses.com>
Date: Thu, 15 Jul 2021 18:20:30 -0500
Subject: [PATCH 45/63] RSOM-324 - update device tree with global CMA phandle
 refs

Add global CMA (contigous memory area) name and USB references
If seems Renesas has modified the UVC driver to prefer these refs for DMA.

Signed-off-by: Charles Stevens <charles.stevens@compasses.com>
---
 .../dts/renesas/beacon-renesom-baseboard.dtsi   | 17 +++++++++++++++++
 .../boot/dts/renesas/beacon-renesom-som.dtsi    |  2 +-
 2 files changed, 18 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
index 386d812f543e..418bb3ac37e6 100755
--- a/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
+++ b/arch/arm64/boot/dts/renesas/beacon-renesom-baseboard.dtsi
@@ -860,8 +860,25 @@
 	pinctrl-0 = <&usb30_pins>;
 	pinctrl-names = "default";
 	status = "okay";
+        memory-region = <&global_cma>;
 };
 
 &msiof_ref_clk {
 	clock-frequency = <50000000>;
 };
+
+&ehci0 {
+        memory-region = <&global_cma>;
+};
+
+&ohci0 {
+        memory-region = <&global_cma>;
+};
+
+&ehci1 {
+        memory-region = <&global_cma>;
+};
+
+&ohci1 {
+        memory-region = <&global_cma>;
+};
diff --git a/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi b/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
index e68e3b9eb3cb..c7052eb9b81d 100644
--- a/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
@@ -27,7 +27,7 @@
 		};
 
 		/* global autoconfigured region for contiguous allocations */
-		linux,cma@58000000 {
+		global_cma: linux,cma@58000000 {
 			compatible = "shared-dma-pool";
 			reusable;
 			reg = <0x00000000 0x58000000 0x0 0x18000000>;
-- 
2.17.1

